packages/notebook/src/actions.tsx

```diff
@@ -2455,6 +2462,27 @@ namespace Private {
         break;
       case 'code':
         if (sessionContext) {
+          if (sessionContext.kernelDisplayStatus === 'initializing' && !initDialogFlag.dialogShown) {
+            initDialogFlag.dialogShown = true;
+            const kernelInitNotificationId = Notification.emit(
+                trans.__(
+                    `Kernel '${sessionContext.kernelDisplayName}' is still initializing, code cells will not be executed.`
```

Comment:
I would advice to display the file path too as a same type of kernel can be used for multiple files; `sessionContext?.path` should provide that information.

Reply:
Is this the file path for the kernel? In most cases, local kernels start almost instantaneously, and the cause we are trying to handle here are remote kernels where the path info might not be very helpful, but I am flexible in this case.

Reply:
It's the file path for the notebook the kernel is executing for